eng/scripts/Generate-Patch.ps1

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+# This script automates the patch release process for a given library.
+# Note that this assumes that the only changes needed in the patch are upgrades to the dependencies.
+# It makes all the necessary changes and pushes them in an upstream branch which can then be used to trigger the release
+# Please follow the necessary steps for the release - validating all the changes, trigger the release pipeline, approve the release etc.
+# It takes in the following arguments
+# The script takes the following arguments
+# 1. ArtifactName - Name of the library that needs to be patched. This is a required parameter.
+# 2. ServiceDirectoryName - The directory name under 'sdk' folder where the source code of the library is present. This is a required parameter.
+# 3. ReleaseVersion - The latest release version of the library which is to be patched. This is a required parameter.
+# 4. PatchVersion - The version of the patch. This is a required parameter.
+#                   TODO: We can remove the need for this parameter by intepreting this from the first version.
+# 5. BranchName - The name of the remote branch where the changes will be pushed. We push the changes to the 'upstream remote'. This is not a required parameter.
+#                 In case the argument is not provided the branch name is release/{ArtifactName}.
+#
+# Example:  .\eng\scripts\Generate-Patch.ps1 -ArtifactName azure-mixedreality-remoterendering -ServiceDirectory remoterendering -ReleaseVersion 1.0.0 -PatchVersion 1.0.1
```

Comment:
I actually started with multiple but it soon became complex. We would have to take multiple versions too. 
Also, in a scenario where we allow multiple artifacts - the order of artifact names changes the output. For example - if common dependency in the first one - it will update all the non-common poms file with its version otherwise not.

Given this I simply avoided added multiple ones. Happy to revisit if you have ideas on how to address these :)

Reply:
Do we need the user to provide patch version? We can infer that from the `-ReleaseVersion`

Reply:
Since both of you had the same comment, I have included the inference of patch version in this PR itself :)